index.rst

```diff
@@ -1,61 +1,486 @@
+The rest of the design described in this document should allow this information to be connected with worker nodes.
+
+Observatory-Butler linkage
+--------------------------
+
+Possibly relevant recommendations:
+
+- *[REC-EXP-1] As planned, program details known to the scheduler (such as science programme and campaign name) should be captured by the Butler.*
+
+As long as these details are included in the FITS headers of the raw files, they can be configured to be capture by the butler and stored in the ``exposure`` dimension table.
+This is already the case for the fields used as examples in this recommendation.
+
+- *[REC-EXP-2] As planned, OCS queue submissions that result in meaningfully grouped observations should be identified as such in the Butler.*
+
+The middleware does currently read a group ID field from the raw headers and store it in the ``exposure`` table.
+We also have a ``visit``-definition scheme that interprets the group ID as a way to relate the snaps that correspond to a single visit, for on-sky science observations only (in the butler dimensions model, a ``visit`` may only be defined for on-sky data).
+
+While that makes this recommendation is formally satisfied, this approach has two (related) problems:
+
+- Using group ID to connect snaps to visits leaves us without a way to create more flexible groups of science exposures.
+- The association of group ID with visits also discourages using group ID for more flexible groups of non-science exposures, such as focus sweeps or flat sequences, or causes confusion about whether visits (or some new visit-like dimension) should be defined and used for these.
+
+It would be much better for a new snap-specific header key to be introduced and interpreted by the middleware, freeing up the group ID to have more flexible and ad-hoc definitions.
+We believe this change is already ticketed in the data acquisition systems, but the work is not yet complete.
+
+Metrics linkage
+---------------
+
+- *[REC-MET-001] For metrics that can be associated with a Butler dataId, the metrics should be persisted using the Data Butler as the source of truth. The dataId associated with the metric should use the full granularity.*
+- *[REC-MET-002] Any system that uses Butler data to derive metrics should persist them in the Butler provided that the metrics are associable with a Data ID.*
+
+This is already implemented and in regular use in the ``faro`` package in particular, and is currently the only way that metrics derived from butler data are initially stored.
+It is arguable whether the butler datasets are considered the source of truth after upload to SQuaSH.
+
+- *[REC-MET-003] When lsst.verify.Job objects are exported, the exported object should include the needed information (run collection and dataId) to associate with the source of truth metric persisted with the Data Butler.*
```

Comment:
I'm assuming "Job" is an overloaded term and has nothing to do with compute jobs in bps.

Reply:
Clarified in the text.